### ISSUE TITLE
[FEATURE] Add golangci-lint Workflow for Automated Code Quality Checks

### ISSUE THREAD

**Issue by kashifkhan0771** [kashifkhan0771/utils#36]
## Feature Description

Introduce a `golangci-lint` workflow to ensure code quality and maintain consistency across the codebase. Using a linter workflow helps detect syntax issues, code style inconsistencies, and potential bugs early, reducing technical debt and improving code readability.

---

## Use Case

- **Maintain Code Quality:** Ensures consistent code style and best practices throughout the project.
- **Catch Potential Issues Early:** Automatically detects possible bugs, dead code, unused imports, and other issues that may go unnoticed in a manual review.
- **Streamline Contributions:** Helps new contributors adhere to established coding conventions by enforcing them at the pull request stage.

---

## Proposed Solution

Add a GitHub Actions workflow that integrates golangci-lint, enabling widely-used linters like govet, gosimple, unused, errcheck, and others relevant to the project. This workflow should run on each pull request and commit to prevent non-compliant code from being merged.

---

## Additional Context

N/A

---

## Pseudo Code

```go
N/A

**Comment by kashifkhan0771** [comment]
If we can configure it to run with a make command locally that would be awesome 😃

**Comment by Kaamkiya** [comment]
I would love to tackle this issue! Please assign me :)

**Comment by kashifkhan0771** [comment]
> By this, do you mean add a Makefile and then call the make command from the workflow?

Yes, the make file is already there. Right now it only has one command `test` to run all test cases. We can configure the linter with it as well so we only need to run command `make lint` and it should check all the code for linters we enable.
